src/app/libp2p_helper/src/libp2p_helper/app.go

```diff
@@ -1090,3 +1310,114 @@ func findPeerInfo(app *app, id peer.ID) (*codaPeerInfo, error) {
 	}
 	return maybePeer, nil
 }
+
+func (app *app) handleIncomingMsg(msg *ipc.Libp2pHelperInterface_Message) {
+	if msg.HasRpcRequest() {
```

Comment:
Can't we use the `msg.Which()` pattern for these unions instead? Then we can clean all of this up by using a jump table to lookup the handlers we want to dispatch to. For example:

```go
func (app *app) handleResetStream(seqno uint64, req ipc.Libp2pHelperInterface_RpcRequest) *capnp.Message {
  m, err := req.ResetStream()
  panicOnErr(err)
  ...
}

type rpcRequestHandler = func(uint64, ipc.Libp2pHelperInterface_RpcRequest) *capnp.Message

var rpcRequestHandlers : map[ipc.Libp2pHelperInterface_RpcRequest_Which]rpcRequestHandler = {
  ipc.Libp2pHelperInterface_RpcRequest_Which_resetStream: handleResetStream,
  ...
}
```

Reply:
Implemented the change, good idea!